manager/dispatcher/dispatcher.go

```diff
@@ -1171,7 +1171,16 @@ func (d *Dispatcher) moveTasksToOrphaned(nodeID string) error {
 		}
 
 		for _, task := range tasks {
-			if task.Status.State < api.TaskStateOrphaned {
+			// Tasks running on an unreachable node need to be marked as
+			// orphaned since we have no idea whether the task is still running
+			// or not.
+			//
+			// This only applies for tasks that could have made progress since
+			// the agent became unreachable (accepted<->running)
+			//
+			// Tasks in a final state (e.g. rejected) *cannot* have made
+			// progress, therefore there's no point in marking them as orphaned
+			if task.Status.State >= api.TaskStateAccepted && task.Status.State <= api.TaskStateRunning {
```

Comment:
I think it should be `api.TaskStateAssigned`. It's dispatcher's responsibility from `Assigned`. See https://github.com/docker/swarmkit/blob/master/manager/dispatcher/dispatcher.go#L51